arcgis-ios-sdk-samples/Edit data/Edit features with feature-linked annotation/EditFeaturesWithFeatureLinkedAnnotationViewController.swift

```diff
@@ -0,0 +1,228 @@
+                // Add the feature layers to the map.
+                map.operationalLayers.add(parcelLinesFeatureLayer)
+                map.operationalLayers.add(addressPointFeatureLayer)
+                // Add the annotation layers to the map.
+                map.operationalLayers.add(parcelLinesAnnotationLayer)
+                map.operationalLayers.add(addressPointsAnnotationLayer)
+            }
+        }
+    }
+    
+    // MARK: - AGSGeoViewTouchDelegate
+    // Select the nearest feature or move the point or polyline vertex to the given screen point.
+    func geoView(_ geoView: AGSGeoView, didTapAtScreenPoint screenPoint: CGPoint, mapPoint: AGSPoint) {
+        // If a feature hasn't been selected.
+        if selectedFeature == nil {
+            selectFeature(screenPoint: screenPoint)
+        } else {
+            // Move the feature.
+            if selectedFeatureIsPolyline {
+                movePolylineVertex(mapPoint: mapPoint)
+            } else {
+                movePoint(mapPoint: mapPoint)
+            }
+        }
+    }
+    
+    func selectFeature(screenPoint: CGPoint) {
+        // Clear any previously selected features.
+        clearSelection()
+        selectedFeature = nil
+        
+        // Identify across all layers
+        mapView.identifyLayers(atScreenPoint: screenPoint, tolerance: 10.0, returnPopupsOnly: false) { (results: [AGSIdentifyLayerResult]?, error: Error?) in
+            if let error = error {
+                self.presentAlert(error: error)
+            } else {
+                results?.forEach { (result) in
+                    if let featureLayer = result.layerContent as? AGSFeatureLayer {
+                        // Get a reference to the identified feature
+                        self.selectedFeature = result.geoElements[0] as? AGSFeature
+                        // If the selected feature is a polyline with any part containing more than one segment (i.e. a curve).
+                        if let polyline = self.selectedFeature?.geometry as? AGSPolyline {
+                            let polylineArray = polyline.parts.array()
+                            polylineArray.forEach { part in
+                                if part.pointCount > 2 {
+                                    // Set selected feature to nil.
+                                    self.selectedFeature = nil
+                                    // Show a message to select straight (single segment) polylines only.
+                                    self.presentAlert(title: "Make a different selection", message: "Select straight (single segment) polylines only.")
+                                    return
+                                }
+                            }
+                        }
+                        // Ensure the feature is not nil and select it.
+                        guard let selectedFeature = self.selectedFeature else { return }
```

Comment:
I put it there because line 84 changes `self.selectedFeature` and was causing complications if I guarded before that. An alternative is guarding before line 86. What're your thoughts?

Reply:
I can move it higher up after line 84 but can't move it higher than that because `self.selectedFeature` is nil prior to that line.

Reply:
Update: I had to leave the guard statement at its original place because of the possible value change of `self.selectedFeature`at line 93. Moving the guard statement prior to that line caused the program to continue with the curved polyline selection and made `selectedFeatureIsPolyline` true when it should be false.